aws-cpp-sdk-transfer/source/transfer/UploadFileRequest.cpp

```diff
@@ -89,52 +92,136 @@ m_getObjectRetries(0),
 m_listObjectsRetries(0),
 m_headBucketRetries(0)
 {
-    if (m_fileStream.good() && m_fileStream.is_open())
+    if (useStream)

```

Comment:
No. You should ALWAYS use stream. If someone passes you a file path, you immediately assign Aws::FStream to it. File Name and IOStream shouldn't exist in the same method. Write one that uses IOStream and another that simply calls it.